tinygrad/tensor.py

```diff
@@ -875,9 +875,9 @@ def sparse_categorical_crossentropy(self, Y:Tensor, ignore_index=-1) -> Tensor:
   # ***** cast ops *****
 
   def cast(self, dtype:DType) -> Tensor:
-    # hack for devices that don't support bfloat16
-    if self.dtype == dtypes.bfloat16:
-      return self.bitcast(dtypes.uint16).cast(dtypes.uint32).mul(1<<16).contiguous().bitcast(dtypes.float32).cast(dtype)
+    # hack loading bf16 from disktensors for devices that don't support bfloat16
```

Comment:
Hmm, this doesn't seem right. This is if the tensor is bfloat16, not being casted to bfloat16

Reply:
then we'll have to extend the hack to linearizer and ops_disk to make the cleanup work.
https://github.com/tinygrad/tinygrad/pull/2794/commits/eead41882342a15bfee2148758818403a6e506e8